uniffi/src/lib.rs

```diff
@@ -347,7 +347,7 @@ unsafe impl ViaFfi for String {
         check_remaining(buf, 4)?;
         let len = usize::try_from(buf.get_i32())?;
         check_remaining(buf, len)?;
-        let bytes = &buf.bytes()[..len];
+        let bytes = &buf.chunk()[..len];
```

Comment:
Hrm...the docs for the [`chunk` method](https://docs.rs/bytes/1.0.1/bytes/trait.Buf.html#method.chunk) note that "this can return shorter slice (this allows non-continuous internal representation)" which IIUC would risk a panic here. In practice though we only ever use this with an underlying `[u8]` which doesn't have a non-continuous internal representation. So I think this is fine, but I'll make a followup to try to make that a little more explicit in the code. Thanks!

Reply:
Is it not the same as before? \
The [previous docs](https://docs.rs/bytes/0.5.6/bytes/trait.Buf.html#tymethod.bytes) have the same warning.

Reply:
I'll checkout your PR and try to decipher your statement, thanks! :)

Reply:
Thanks for checking - it seems very likely that the old version also had this behavior and I just didn't notice!

Reply:
FWIW, if you do take a look, I'm curious whether the commentary I added in https://github.com/mozilla/uniffi-rs/pull/455 made any sense.

Reply:
After navigating a bit through the rest of the library, it does. \
Thanks!